app/src/main/java/org/projectbuendia/client/ui/chart/PatientChartActivity.java

```diff
@@ -351,6 +381,44 @@ public void onClick(View v) {
         });
     }
 
+    /**
+     *  Set the margin_top to a percentage of the height of the root view. Here's why:
+     *  - If we set a layout_weight on the notes panel, we can ensure that the notes panel
+     *    takes up that fraction of the height of its parent.
+     *  - When the keyboard is displayed, however, the panel height adjusts to (weight *
+     *    new parent height).
+     *  - For small values of layout_weight (e.g. < 0.5) this makes the notes panel noticeably
+     *    larger when the keyboard is expanded.
+     *  - So we compute the height at a time when we know the keyboard isn't showing, and assign
+     *    (weight * parent height) to the margin, so that much of the underlying view will always
+     *    be visible.
+     *
+     * @param panelHeightFrac the height to which the notes panel should be set at.
+     */
+    private void setNotesPanelMaxHeightToFracOfParent(final float panelHeightFrac) {
```

Comment:
Android layout quirks. There isn't a way of doing this using %? (Just out of curiosity)